accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,503 @@
+
+_**Open Issue**. Will the above scheme work for iPad?_
+
+### Mapping to properties
+
+There are three existing MSBuild properties:
+
+| Property                        | Meaning            | Examples                         |
+|---------------------------------|--------------------|----------------------------------|
+| TargetFramework (TFM)           | The friendly name  | `net4`, `netcoreapp3.0`          |
+| TargetFrameworkIdentifier (TFI) | The long name      | `.NETFramework` or `.NETCoreApp` |
+| TargetFrameworkVersion (TFV)    | The version number | `2`, `3.0`, `3.1`                |
+| TargetFrameworkProfile (TFP)    | The profile        | `Client` or `Profile124`         |
+
+_**Open Issue**. The SDK has this logic duplicated from NuGet because they need
+to do this during evaluation where they can’t call custom targets. We could make
+this an MSBuild intrinsic, but that seems like a lot of work. Maybe we just live
+with the duplication. But bottom line is that we need to make that change in
+MSBuild too._
+
+* [@rainersigwald](https://github.com/rainersigwald): We've made it pretty far
+  with duplicated logic, but it results in really ugly MSBuild, since it's not a
+  very expressive programming language. It also creates the potential for drift
+  between the two definitions. That said, exposing it directly would create a
+  new tight coupling between MSBuild and NuGet that hasn't historically existed.
+  It would probably require a direct dependency and update flow plus coherency
+  requirements on both .NET Core SDK and VS insertions. If the logic were in a
+  separate package (we've talked about pushing it down to the framework at
+  various times) it'd be great to just expose that. With the logic in NuGet,
+  it's reasonable either way, just different tradeoffs. I'm amenable to exposing
+  a property function, but maybe we should go down the road of not doing it at
+  first.
+
+We're going to map the new entries as follows:
+
+| Framework      | Identifier    | Version | Profile |
+|----------------|---------------|---------|---------|
+| net48          | .NETFramework | 4.8     |         |
+| net5.0         | .NETCoreApp   | 5.0     |         |
+| net5.0-android | .NETCoreApp   | 5.0     | android |
+| net5.0-ios     | .NETCoreApp   | 5.0     | ios     |
+| net5.0-win     | .NETCoreApp   | 5.0     | win     |
+
+_**Open Issue**. Please note that `net5.0`+ will map the TFI to `.NETCoreApp`.
+We need to announce this change so that package authors with custom .props and
+.targets are prepared. Link to DavKean’s doc on how to do it._
+
+_**Open Issue**. We should try to keep the TFI out of the .nuspec file. It seems
+NuGet uses the long form `.NETFramework,Version=4.5` in the dependency groups.
+We may want to change NuGet to allow the friendly name there as well and update
+our packaging tools to re-write to friendly name on pack._
+
+Specifically:
+
+* **We'll continue to use .NETCoreApp as the TFI**. This reduces the number of
```

Comment:
Seems a shame to be stuck with a confusing legacy name. How widely is the string ".NETCoreApp" used in the SDK?

Reply:
It's gonna cause a whole lot of confusion. It's better to remove the whole `TargetFramework` concept altogether. Just go with `Runtime` (`Platform`/`Machine`). It's much simpler and easy to understand. My examples are not a definitive design but it can go from there...

My solution is simple. As long as you target specific `Runtime` (`Platform`/`Machine`), the libs produced will work on that. The `TargetFrameworkVersion` will be there as the first two version numbers (major, minor) in `RuntimeVersion` if we follow Semantic Versioning strictly.

See https://github.com/dotnet/designs/pull/92#issuecomment-587106124

Reply:
We discussed this and concluded that keeping the current TFI is the best solution.